python/cudf/cudf/core/dataframe.py

```diff
@@ -5652,20 +5652,35 @@ def cov(self, **kwargs):
         df._set_column_names_like(self)
         return df
 
-    @_cudf_nvtx_annotate
-    def corr(self):
-        """Compute the correlation matrix of a DataFrame."""
-        corr = cupy.corrcoef(self.values, rowvar=False)
-        cols = self._data.to_pandas_index()
-        df = DataFrame(cupy.asfortranarray(corr)).set_index(cols)
-        df._set_column_names_like(self)
+    def corr(self, method="pearson"):
+        """Compute the correlation matrix of a DataFrame.
+        Parameters
+        ----------
+        method : {'pearson', 'spearman'}, default 'pearson'
+            The correlation method to use, one of 'pearson' or 'spearman'.
+
+        Returns
+        -------
+        DataFrame
+            The requested correlation matrix.
+        """
+        if method == "pearson":
+            corr = cupy.corrcoef(self.values, rowvar=False)
+            cols = self._data.to_pandas_index()
+            df = DataFrame(cupy.asfortranarray(corr)).set_index(cols)
+            df._set_column_names_like(self)
+        elif method == "spearman":
+            corr = cupy.corrcoef(self.rank().values, rowvar=False)
+            df = DataFrame(cupy.asfortranarray(corr)).set_index(self._column_names)
+            df.columns = self.columns
+        else:
+            raise ValueError("method must be either 'pearson', 'spearman'")
```

Comment:
This part may better be implemented this way to reduce duplication.
```suggestion
        if method == "pearson":
            values = self.values
        elif method == "spearman":
            values = self.rank().values
        else:
            raise ValueError("method must be either 'pearson', 'spearman'")
        corr = cupy.corrcoef(values, rowvar=False)
        cols = self._data.to_pandas_index()
        df = DataFrame(cupy.asfortranarray(corr)).set_index(cols)
        df._set_column_names_like(self)
```